src/txn/txn_recover.c

```diff
@@ -979,6 +979,16 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
          */
         WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
+    /* Initialize the connection's base write generation after rollback to stable. */
```

Comment:
As of now metadata don't use history store and don't write unstable updates to disk, so there is no problem in resetting the transaction ids for the metadata. But due to the increase of connection writegen, it is possible that history store transaction data may get lost when it gets accessed during log replay. So I feel move the RTS to a location before log replay?

Reply:
Moved the RTS mechanism before log recovery.